test/integration/batcher/TwoWayBatcherIntegration.test.ts

```diff
@@ -0,0 +1,609 @@
+          .withArgs(utils.parseEther('100').add(1))
+      })
+
+      it('repays token partial rounding', async () => {
+        const { user } = proto
+
+        await expect(batcher.connect(user).withdraw(utils.parseEther('50').add(1)))
+          .to.be.revertedWithCustomError(batcher, 'TwoWayBatcherInvalidTokenAmount')
+          .withArgs(utils.parseEther('50').add(1))
+      })
+
+      it('repays token multiple users', async () => {
+        const { usdc, user, user2 } = proto
+
+        await batcher.connect(user2).deposit(utils.parseEther('200'))
+
+        await expect(batcher.connect(user).withdraw(utils.parseEther('100')))
+          .to.emit(batcher, 'Withdraw')
+          .withArgs(user.address, utils.parseEther('100'))
+        await expect(batcher.connect(user2).withdraw(utils.parseEther('101')))
+          .to.emit(batcher, 'Withdraw')
+          .withArgs(user2.address, utils.parseEther('101'))
+
+        expect(await batcher.totalSupply()).to.equal(utils.parseEther('99'))
+        expect(await batcher.balanceOf(user.address)).to.equal(0)
+        expect(await batcher.balanceOf(user2.address)).to.equal(utils.parseEther('99'))
+
+        expect(await usdc.balanceOf(batcher.address)).to.equal(99_000_000)
+        expect(await usdc.balanceOf(user.address)).to.equal(1_000_000_000_000)
+        expect(await usdc.balanceOf(user2.address)).to.equal(999_901_000_000)
+      })
+
+      context('rebalance required', () => {
+        beforeEach(async () => {
+          const { dsu, usdc, reserve, user2 } = proto
+          // Get DSU from reserve and unwrap via batcher
+          await usdc.connect(user2).approve(reserve.address, 100_000_000)
+          await dsu.connect(user2).approve(batcher.address, utils.parseEther('100'))
+          await reserve.connect(user2).mint(utils.parseEther('100'))
+          await batcher.connect(user2).unwrap(utils.parseEther('100'), user2.address)
+        })
+
+        it('repays token full', async () => {
+          const { usdc, user } = proto
+
+          await expect(batcher.connect(user).withdraw(utils.parseEther('100')))
+            .to.emit(batcher, 'Withdraw')
+            .withArgs(user.address, utils.parseEther('100'))
+            .to.emit(batcher, 'Rebalance')
+            .withArgs(0, utils.parseEther('100'))
+
+          expect(await batcher.totalSupply()).to.equal(0)
+          expect(await batcher.balanceOf(user.address)).to.equal(0)
+
+          expect(await usdc.balanceOf(batcher.address)).to.equal(0)
+          expect(await usdc.balanceOf(user.address)).to.equal(1_000_000_000_000)
+        })
+
+        it('repays token partial', async () => {
+          const { usdc, user } = proto
+
+          await expect(batcher.connect(user).withdraw(utils.parseEther('51')))
+            .to.emit(batcher, 'Withdraw')
+            .withArgs(user.address, utils.parseEther('51'))
+            .to.emit(batcher, 'Rebalance')
```

Comment:
did we need to split these out re: waffle issue?

Reply:
Nah, I updated this repo to use `hardhat-chai-matchers` which fixes this issue. Confirmed the fix works as well